integration-tests/opentelemetry-mongodb-client-instrumentation/src/main/java/io/quarkus/it/opentelemetry/ReactiveBookResource.java

```diff
@@ -38,6 +40,14 @@ public CompletionStage<List<Book>> getBooks() {
         return getCollection().find().collect().asList().subscribeAsCompletionStage();
     }
 
+    @GET
+    @Path("/invalid")
+    public CompletionStage<List<Book>> getBooksError() {
+        BsonDocument query = new BsonDocument();
+        query.put("$invalidop", new BsonDouble(0d));
+        return getCollection().find(query).collect().asList().subscribeAsCompletionStage();
```

Comment:
The context propagation will not happen when calling the MongoDB directly. 
Please check the injected reactive client, it should provide the required propagation: https://quarkus.io/guides/mongodb#reactive.

Reply:
I've fixed reactive context propagation by a registering `com.mongodb.reactivestreams.client.ReactiveContextProvider`. The same should be done also for blocking client when mongo queries are executed in a thread pool, but thats would be another PR. My first attempt to implement `com.mongodb.client.SynchronousContextProvider` did not work, unfortunately